### ISSUE TITLE
Handlers: Message get formatted twice

### ISSUE THREAD

**Issue by MineTheCube** [kangarko/ChatControl#9]
I think bug came with 5.1.7, as I never noticed it before. So when a handler is called, the message seems to be formatted twice.

File **rules.txt**:
```
match \b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}(\b|:)
handle as advertisement
```

File **handlers.yml**:
```
advertisement:
  Bypass_With_Permission: "chatcontrol.bypass.ad"
  Ignored_In_Commands:
    - // # WorldEdit
  Player_Warn_Message: "Advertising is forbidden !"
  Console_Message: "&c&lPUB &e%player &7>&f %message"
  Staff_Alert_Message: "&c&lPUB &e%player &7>&f %message"
  Staff_Alert_Permission: chatcontrol.notify.ad
  Block_Message: true
```

**Result**: (`PUB` means `advertising`)

![](http://i.imgur.com/rUidjzM.png)
![](http://i.imgur.com/cIfZ9PI.png)

**Comment by kangarko** [comment]
1. Do you have Chat Formatter enabled? If so please post me its config from settings.yml
2. Does this happen also when normal chatting, e.g. messages not cough by any rule? If so, do you have any other plugin for chat formatting? If so, please post /vault-info input.

**Comment by kangarko** [comment]
Try changing following keys:
>  Console_Message: "&c&lPUB &e%player &7>&f %message"
>  Staff_Alert_Message: "&c&lPUB &e%player &7>&f %message"
to

> Console_Message: "&e%player &7>&f %message"
> Staff_Alert_Message: &e%player &7>&f %message"

**Comment by MineTheCube** [comment]
*Do you have Chat Formatter enabled in ChatControl?* Nope.

*Any other chat formatting plugin ?* Mine, but even disabled the bug still occur.

![](http://i.imgur.com/NFavJVS.png)

Btw, I'm using ChatControl v5.1.8. And I made few tests, see below.

--------

```
swear:
  Bypass_With_Permission: "chatcontrol.bypass.swear"
  Player_Warn_Message: "Swearing is forbidden !"
  Staff_Alert_Message: "%player - %message"
  Staff_Alert_Permission: chatcontrol.notify.swear
  Block_Message: true
```
![](http://i.imgur.com/JMZpOGT.png)

--------

```
swear:
  Bypass_With_Permission: "chatcontrol.bypass.swear"
  Player_Warn_Message: "Swearing is forbidden !"
  Staff_Alert_Message: "- %message"
  Staff_Alert_Permission: chatcontrol.notify.swear
  Block_Message: true
```
![](http://i.imgur.com/PLaQWBp.png)

--------
```
swear:
  Bypass_With_Permission: "chatcontrol.bypass.swear"
  Player_Warn_Message: "Swearing is forbidden !"
  Staff_Alert_Message: "%player -"
  Staff_Alert_Permission: chatcontrol.notify.swear
  Block_Message: true
```
![](http://i.imgur.com/bxCrFYI.png)

--------

Happy debbuging :)

**Comment by dartasen** [comment]
I have the same problem on Spigot 1.8.8 with the latest version of ChatControl ^^

**Comment by kangarko** [comment]
Hey, this definitely seems like issue with the plugin. I am sorry and I will fix it asap for me.

**Comment by kangarko** [comment]
Issue with message being formatted twice has been fixed in v5.1.9
